libsplinter/src/biome/rest_api/resources/key_management.rs

```diff
@@ -26,3 +28,22 @@ pub(crate) struct UpdatedKey {
     pub public_key: String,
     pub new_display_name: String,
 }
+
+#[derive(Serialize, Clone)]
+pub(crate) struct ResponseKey {
```

Comment:
The same pattern could apply here:
```rust
struct ResponseKey<'a>{
    public_key: &'a str,
    ...
}
```

Given the `From` implementation, you probably don't need the fields to be `pub`, either.

Reply:
I don't think it's possible to implement this one using the same pattern, as the `key` object used to fill in the values for the ResponseKey is being consumed before the ResponseKey is returned. The fields probably don't have to be pub, though.